NTRU doesn't compile with GCC 11

As identified in https://github.com/open-quantum-safe/liboqs/issues/1035, NTRU doesn't compile with GCC 11 due to mismatching function signatures:

```
FAILED: src/kem/ntru/CMakeFiles/ntru_hps2048677_avx2.dir/pqclean_ntruhps2048677_avx2/pack3.c.o
/usr/bin/cc  -Iinclude -I../src/kem/ntru/pqclean_ntruhps2048677_avx2 -I../src/common/pqclean_shims -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -O3 -fomit-frame-pointer -fdata-sections -ffunction-sections -Wl,--gc-sections -mavx2 -mbmi2 -std=gnu11 -MD -MT src/kem/ntru/CMakeFiles/ntru_hps2048677_avx2.dir/pqclean_ntruhps2048677_avx2/pack3.c.o -MF src/kem/ntru/CMakeFiles/ntru_hps2048677_avx2.dir/pqclean_ntruhps2048677_avx2/pack3.c.o.d -o src/kem/ntru/CMakeFiles/ntru_hps2048677_avx2.dir/pqclean_ntruhps2048677_avx2/pack3.c.o -c ../src/kem/ntru/pqclean_ntruhps2048677_avx2/pack3.c
../src/kem/ntru/pqclean_ntruhps2048677_avx2/pack3.c:3:64: error: argument 1 of type ‘unsigned char[272]’ with mismatched bound [-Werror=array-parameter=]
    3 | void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
      |                                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/kem/ntru/pqclean_ntruhps2048677_avx2/pack3.c:1:
../src/kem/ntru/pqclean_ntruhps2048677_avx2/poly.h:25:64: note: previously declared as ‘unsigned char[136]’
   25 | void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a);
      |                                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/kem/ntru/pqclean_ntruhps2048677_avx2/pack3.c:24:81: error: argument 2 of type ‘const unsigned char[272]’ with mismatched bound [-Werror=array-parameter=]
   24 | void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
      |                                                             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/kem/ntru/pqclean_ntruhps2048677_avx2/pack3.c:1:
../src/kem/ntru/pqclean_ntruhps2048677_avx2/poly.h:26:81: note: previously declared as ‘const unsigned char[136]’
   26 | void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]);
      |                                                             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

and also:

```
../src/kem/ntru/pqclean_ntruhps4096821_avx2/pack3.c:3:64: error: argument 1 of type 'unsigned char[328]' with mismatched bound [-Werror=array-parameter=]
    3 | void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
      |                                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/kem/ntru/pqclean_ntruhps4096821_avx2/pack3.c:1:
../src/kem/ntru/pqclean_ntruhps4096821_avx2/poly.h:25:64: note: previously declared as 'unsigned char[164]'
   25 | void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a);
      |                                                  ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/kem/ntru/pqclean_ntruhps4096821_avx2/pack3.c:17:81: error: argument 2 of type 'const unsigned char[328]' with mismatched bound [-Werror=array-parameter=]
   17 | void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
      |                                                             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/kem/ntru/pqclean_ntruhps4096821_avx2/pack3.c:1:
../src/kem/ntru/pqclean_ntruhps4096821_avx2/poly.h:26:81: note: previously declared as 'const unsigned char[164]'
   26 | void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]);
      |                                                             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

and possibly elsewhere.

Tagging @jschanck

--- a/crypto_kem/ntruhps2048509/META.yml
+++ b/crypto_kem/ntruhps2048509/META.yml
@@ -23,9 +23,9 @@ auxiliary-submitters:
   - Zhenfei Zhang
 implementations:
     - name: clean
-      version: https://github.com/jschanck/ntru/tree/a43a4457 reference implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 reference implementation
     - name: avx2
-      version: https://github.com/jschanck/ntru/tree/a43a4457 avx2 implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 avx2 implementation
       supported_platforms:
           - architecture: x86_64
             operating_systems:

--- a/crypto_kem/ntruhps2048509/avx2/pack3.c
+++ b/crypto_kem/ntruhps2048509/avx2/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS2048509_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS2048509_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
     int j;
@@ -21,7 +21,7 @@ void PQCLEAN_NTRUHPS2048509_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MS
     msg[i] = c;
 }
 
-void PQCLEAN_NTRUHPS2048509_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS2048509_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
     int j;

--- a/crypto_kem/ntruhps2048509/clean/pack3.c
+++ b/crypto_kem/ntruhps2048509/clean/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS2048509_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS2048509_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
     int j;
@@ -21,7 +21,7 @@ void PQCLEAN_NTRUHPS2048509_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_M
     msg[i] = c;
 }
 
-void PQCLEAN_NTRUHPS2048509_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS2048509_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
     int j;

--- a/crypto_kem/ntruhps2048677/META.yml
+++ b/crypto_kem/ntruhps2048677/META.yml
@@ -23,9 +23,9 @@ auxiliary-submitters:
   - Zhenfei Zhang
 implementations:
     - name: clean
-      version: https://github.com/jschanck/ntru/tree/a43a4457 reference implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 reference implementation
     - name: avx2
-      version: https://github.com/jschanck/ntru/tree/a43a4457 avx2 implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 avx2 implementation
       supported_platforms:
           - architecture: x86_64
             operating_systems:

--- a/crypto_kem/ntruhps2048677/avx2/pack3.c
+++ b/crypto_kem/ntruhps2048677/avx2/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
     int j;
@@ -21,7 +21,7 @@ void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MS
     msg[i] = c;
 }
 
-void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS2048677_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
     int j;

--- a/crypto_kem/ntruhps2048677/clean/pack3.c
+++ b/crypto_kem/ntruhps2048677/clean/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS2048677_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS2048677_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
     int j;
@@ -21,7 +21,7 @@ void PQCLEAN_NTRUHPS2048677_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_M
     msg[i] = c;
 }
 
-void PQCLEAN_NTRUHPS2048677_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS2048677_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
     int j;

--- a/crypto_kem/ntruhps4096821/META.yml
+++ b/crypto_kem/ntruhps4096821/META.yml
@@ -23,9 +23,9 @@ auxiliary-submitters:
   - Zhenfei Zhang
 implementations:
     - name: clean
-      version: https://github.com/jschanck/ntru/tree/a43a4457 reference implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 reference implementation
     - name: avx2
-      version: https://github.com/jschanck/ntru/tree/a43a4457 avx2 implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 avx2 implementation
       supported_platforms:
           - architecture: x86_64
             operating_systems:

--- a/crypto_kem/ntruhps4096821/avx2/pack3.c
+++ b/crypto_kem/ntruhps4096821/avx2/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
 
@@ -14,7 +14,7 @@ void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MS
     }
 }
 
-void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS4096821_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
 

--- a/crypto_kem/ntruhps4096821/clean/pack3.c
+++ b/crypto_kem/ntruhps4096821/clean/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHPS4096821_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHPS4096821_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
 
@@ -14,7 +14,7 @@ void PQCLEAN_NTRUHPS4096821_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_M
     }
 }
 
-void PQCLEAN_NTRUHPS4096821_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHPS4096821_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
 

--- a/crypto_kem/ntruhrss701/META.yml
+++ b/crypto_kem/ntruhrss701/META.yml
@@ -23,9 +23,9 @@ auxiliary-submitters:
   - Zhenfei Zhang
 implementations:
     - name: clean
-      version: https://github.com/jschanck/ntru/tree/a43a4457 reference implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 reference implementation
     - name: avx2
-      version: https://github.com/jschanck/ntru/tree/a43a4457 avx2 implementation
+      version: https://github.com/jschanck/ntru/tree/3227cd42 avx2 implementation
       supported_platforms:
           - architecture: x86_64
             operating_systems:

--- a/crypto_kem/ntruhrss701/avx2/pack3.c
+++ b/crypto_kem/ntruhrss701/avx2/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHRSS701_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHRSS701_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
 
@@ -14,7 +14,7 @@ void PQCLEAN_NTRUHRSS701_AVX2_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBY
     }
 }
 
-void PQCLEAN_NTRUHRSS701_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHRSS701_AVX2_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
 

--- a/crypto_kem/ntruhrss701/clean/pack3.c
+++ b/crypto_kem/ntruhrss701/clean/pack3.c
@@ -1,6 +1,6 @@
 #include "poly.h"
 
-void PQCLEAN_NTRUHRSS701_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGBYTES], const poly *a) {
+void PQCLEAN_NTRUHRSS701_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_PACK_TRINARY_BYTES], const poly *a) {
     int i;
     unsigned char c;
 
@@ -14,7 +14,7 @@ void PQCLEAN_NTRUHRSS701_CLEAN_poly_S3_tobytes(unsigned char msg[NTRU_OWCPA_MSGB
     }
 }
 
-void PQCLEAN_NTRUHRSS701_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_OWCPA_MSGBYTES]) {
+void PQCLEAN_NTRUHRSS701_CLEAN_poly_S3_frombytes(poly *r, const unsigned char msg[NTRU_PACK_TRINARY_BYTES]) {
     int i;
     unsigned char c;
 

--- a/test/test_functest.py
+++ b/test/test_functest.py
@@ -76,7 +76,10 @@ def test_functest_sanitizers(implementation, impl_path, test_dir,
                  IMPLEMENTATION=implementation.name,
                  EXTRAFLAGS=(
                      '-g -fsanitize=address,undefined '
-                     '-fno-sanitize-recover=undefined'),
+                     '-fno-sanitize-recover=undefined '
+                     # TODO(JMS) Remove explicit -latomic if/when gcc fixes:
+                     # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81358
+                     '-Wno-unused-command-line-argument -latomic'),
                  SCHEME_DIR=impl_path,
                  DEST_DIR=dest_dir,
                  working_dir=os.path.join(test_dir, 'test'),

